src/app/app.component.ts

```diff
@@ -45,18 +49,34 @@ export class AppComponent implements OnInit, OnDestroy {
 
     // 3. Add debounce to prevent API calls until user stop typing.
 
-    this.charactersResults$ = this.searchTermByCharacters
-        .pipe
-        // YOUR CODE STARTS HERE
+    this.charactersResults$ = this.searchTermByCharacters.pipe(
+      // YOUR CODE STARTS HERE
+      debounceTime(300),
+      distinctUntilChanged(),
+      filter(inputValue => inputValue.length >= 3),
+      switchMap(inputValue => this.mockDataService.getCharacters(inputValue))
+    );
 
-        // YOUR CODE ENDS HERE
-        ();
+    this.subscriptions.push(this.charactersResults$.subscribe());
+    // YOUR CODE ENDS HERE
   }
 
   loadCharactersAndPlanet(): void {
     // 4. On clicking the button 'Load Characters And Planets', it is necessary to process two requests and combine the results of both requests into one result array. As a result, a list with the names of the characters and the names of the planets is displayed on the screen.
     // Your code should looks like this: this.planetAndCharactersResults$ = /* Your code */
     // YOUR CODE STARTS HERE
+    const characters$ = this.mockDataService.getCharacters();
+    const planets$ = this.mockDataService.getPlatents();
+
+    this.planetAndCharactersResults$ = forkJoin([characters$, planets$]).pipe(
+      map(([characters, planets]) => [...characters, ...planets]),
+      catchError(error => {
+        console.error('Error fetching data:', error);
+        return throwError(error);
+      })
+    );
+
+    this.subscriptions.push(this.planetAndCharactersResults$.subscribe());
```

Comment:
You don't need to subscribe/unsubscribe to/from planetAndCharactersResults$ in the component, because your template uses an async pipe. It will subscribe/unsubscribe automatically.